server/methods/accounts/accounts.js

```diff
@@ -517,13 +530,30 @@ export function addressBookUpdate(address, accountUserId, type) {
     accountsUpdateQuery.$set.name = address.fullName;
   }
 
+  // Update the Meteor.users collection with new address info
   Meteor.users.update(Meteor.userId(), userUpdateQuery);
 
+  // Update the Reaction Accounts collection with new address info
   const updatedAccount = Accounts.update({
     userId,
     "profile.addressBook._id": address._id
   }, accountsUpdateQuery);
```

Comment:
This is OK, but it wouldn't be that hard to eliminate the earlier two `Accounts.update` calls entirely. It already starts with a `findOne` setting `account`. So in place of the earlier `Accounts.update` calls, you can loop through all of `account.profile.addressBook` array and set every `true` isBillingDefault or `isShippingDefault` to `false`. Then down here, replace one of the `account.profile.addressBook` items with the updated `address` based on its ID.

Then you only need one final actual DB update, doing `{ $set: { "profile.addressBook": account.profile.addressBook } }`